app/browser/reducers/sitesReducer.js

```diff
@@ -49,28 +49,32 @@ const sitesReducer = (state, action, immutableAction) => {
       }
       break
     case appConstants.APP_ADD_SITE:
+      const isSyncEnabled = syncEnabled()
       if (Immutable.List.isList(action.siteDetail)) {
         action.siteDetail.forEach((s) => {
           state = siteUtil.addSite(state, s, action.tag, undefined, action.skipSync)
+          if (isSyncEnabled) {
+            state = syncUtil.updateSiteCache(state, s)
+          }
         })
       } else {
         let sites = state.get('sites')
         if (!action.siteDetail.get('folderId') && siteUtil.isFolder(action.siteDetail)) {
           action.siteDetail = action.siteDetail.set('folderId', siteUtil.getNextFolderId(sites))
         }
         state = siteUtil.addSite(state, action.siteDetail, action.tag, action.originalSiteDetail, action.skipSync)
-      }
-      if (action.destinationDetail) {
-        const sourceKey = siteUtil.getSiteKey(action.siteDetail)
-        const destinationKey = siteUtil.getSiteKey(action.destinationDetail)
+        if (action.destinationDetail) {
```

Comment:
all of this code seems to assume that action.siteDetail is not a List, so i think it should go within the `if` block instead of outside it